src/gcp/artifactregistry.ts

```diff
@@ -21,8 +39,71 @@ export interface Operation {
   };
 }
 
+export interface CleanupPolicyCondition {
+  tagState: string;
+  olderThan: string;
+  packageNamePrefixes?: string[];
+  tagPrefixes?: string[];
+  versionNamePrefixes?: string[];
+  newerThan?: string;
+}
+
+export interface CleanupPolicyMostRecentVersions {
+  packageNamePrefixes?: string[];
+  keepCount: number;
+}
+
+export interface CleanupPolicy {
+  id: string;
+  action: string;
+  condition?: CleanupPolicyCondition;
+  mostRecentVersions?: CleanupPolicyMostRecentVersions;
+}
+
 /** Delete a package. */
 export async function deletePackage(name: string): Promise<Operation> {
   const res = await client.delete<Operation>(name);
   return res.body;
 }
+
+/**
+ * Get a repository from Artifact Registry.
+ */
+export async function getRepository(repoPath: string): Promise<Repository> {
+  const res = await client.get<Repository>(repoPath);
+  return res.body;
+}
+
+/**
+ * Update an Artifact Registry repository.
+ */
+export async function updateRepository(repo: Partial<Repository>): Promise<Repository> {
```

Comment:
FYI: I've improved the state of the art here on what you can do instead of Partials.

```ts
type RepositoryOutputOnlyFields = 'foo', 'bar.baz';
assertImplements<RepsotoryOutputOnlyFields, DeepKeysOf<Repository>>();

// ...
export async function updateRepository(repo: DeepOmit<Repository, RepositoryOutputOnlyFields>): Promise<Repository> {
```

Reply:
Very neato.